lib/osv/models.py

```diff
@@ -336,11 +336,17 @@ def _pre_put_hook(self):
     })
     self.project.sort()
 
-    self.ecosystem = list({
-        pkg.package.ecosystem
+    ecosystems_set = {
+        pkg.package.ecosystem.split(':')
         for pkg in self.affected_packages
         if pkg.package.ecosystem
-    })
+    }
+
+    # For all ecosystems that specify a specific version with colon,
+    # also add the base name
+    ecosystems_set.update({x.split(':')[0] for x in ecosystems_set if ':' in x})
```

Comment:
We can skip the `if ':' in x`, since split() will always at least return an array of size 1.